components/plugin_marketplace/marketplace_item/marketplace_item.js

```diff
@@ -4,16 +4,208 @@
+export const UpdateVersion = ({version, releaseNotesUrl}) => {
+    if (!releaseNotesUrl) {
+        return version;
+    }
+
+    return (
+        <a
+            target='_blank'
+            rel='noopener noreferrer'
+            href={releaseNotesUrl}
+        >
+            {version}
+        </a>
+    );
+};
+
+UpdateVersion.propTypes = {
+    version: PropTypes.string.isRequired,
+    releaseNotesUrl: PropTypes.string,
+};
+
+// UpdateDetails renders an inline update prompt for plugins, when available.
+export const UpdateDetails = ({version, releaseNotesUrl, installedVersion, isInstalling, onUpdate}) => {
+    if (!installedVersion || isInstalling) {
+        return null;
+    }
+
+    var isUpdate = false;
+    try {
+        isUpdate = semver.gt(version, installedVersion);
+    } catch (e) {
+        // If we fail to parse the version, assume not an update;
+    }
+
+    if (!isUpdate) {
+        return null;
+    }
+
+    return (
+        <div className={classNames('update')}>
+            <FormattedMessage
+                id='marketplace_modal.list.update_available'
+                defaultMessage='Update available:'
+            />
+            {' '}
+            <UpdateVersion
+                version={version}
+                releaseNotesUrl={releaseNotesUrl}
+            />
+            {' - '}
+            <b>
+                <a onClick={onUpdate}>
+                    <FormattedMessage
+                        id='marketplace_modal.list.update'
+                        defaultMessage='Update'
+                    />
+                </a>
+            </b>
+        </div>
+    );
+};
+
+UpdateDetails.propTypes = {
+    version: PropTypes.string.isRequired,
+    releaseNotesUrl: PropTypes.string,
+    installedVersion: PropTypes.string,
+    isInstalling: PropTypes.bool.isRequired,
+    onUpdate: PropTypes.func.isRequired,
+};
+
+// UpdateConfirmationModal prompts before allowing upgrade, specially handling major version changes.
+export const UpdateConfirmationModal = ({show, name, version, installedVersion, releaseNotesUrl, onUpdate, onCancel}) => {
+    if (!installedVersion) {
+        return null;
+    }
+
+    var isUpdate = false;
+    try {
+        isUpdate = semver.gt(version, installedVersion);
+    } catch (e) {
+        // If we fail to parse the version, assume not an update;
+    }
+
+    if (!isUpdate) {
+        return null;
+    }
+
+    const messages = [(
+        <p key='intro'>
+            <FormattedMessage
+                id='marketplace_modal.list.update_confirmation.message.intro'
+                defaultMessage={`Are you sure you want to update the ${name} plugin to ${version}?`}
+                values={{name, version}}
+            />
+        </p>
+    )];
+
+    if (releaseNotesUrl) {
+        messages.push(
+            <p key='current'>
+                <FormattedHTMLMessage
+                    id='marketplace_modal.list.update_confirmation.message.current_with_release_notes'
```

Comment:
This is missing in `i18n/en.json`. It should be there, right?

Reply:
Indeed, not sure why mmjstool isn't extracting. Will fix manually.